code/modules/abstract/abstract_airlock_marker.dm

```diff
@@ -0,0 +1,175 @@
+#define ATAG_INT_DOOR       BITFLAG(6)
+/// Interior placed air sensor.
+#define ATAG_INT_SENSOR     BITFLAG(7)
+/// Interior placed pump inlet/outlet.
+#define ATAG_INT_PUMP_OUT   BITFLAG(8)
+
+// Use this macro to declare the required types for mapping your airlock.
+// Ex. DECLARE_AIRLOCK_HELPER_TYPES(example, "example airlock", (ATAG_AIR_PUMP|ATAG_SENSOR_CHAMBER|ATAG_EXT_DOOR|ATAG_INT_DOOR))
+#define DECLARE_AIRLOCK_HELPER_TYPES(AIRLOCK_ID, AIRLOCK_NAME, AIRLOCK_FLAGS)                                               \
+/obj/abstract/airlock_marker/airlock_controller/##AIRLOCK_ID/name         = "##AIRLOCK_NAME docking controller marker"; \
+/obj/abstract/airlock_marker/airlock_controller/##AIRLOCK_ID/id_tag_flags = AIRLOCK_FLAGS; \
+/obj/abstract/airlock_marker/airlock_controller/##AIRLOCK_ID/airlock_id   = "##AIRLOCK_ID"; \
+/obj/abstract/airlock_marker/sensor/chamber/##AIRLOCK_ID/name             = "##AIRLOCK_NAME chamber sensor marker"; \
+/obj/abstract/airlock_marker/sensor/chamber/##AIRLOCK_ID/airlock_id       = "##AIRLOCK_ID"; \
+/obj/abstract/airlock_marker/sensor/internal/##AIRLOCK_ID/name            = "##AIRLOCK_NAME internal sensor marker"; \
+/obj/abstract/airlock_marker/sensor/internal/##AIRLOCK_ID/airlock_id      = "##AIRLOCK_ID"; \
+/obj/abstract/airlock_marker/sensor/external/##AIRLOCK_ID/name            = "##AIRLOCK_NAME external sensor marker"; \
+/obj/abstract/airlock_marker/sensor/external/##AIRLOCK_ID/airlock_id      = "##AIRLOCK_ID"; \
+/obj/abstract/airlock_marker/pump/chamber/##AIRLOCK_ID/name               = "##AIRLOCK_NAME chamber pump marker"; \
+/obj/abstract/airlock_marker/pump/chamber/##AIRLOCK_ID/airlock_id         = "##AIRLOCK_ID"; \
+/obj/abstract/airlock_marker/pump/internal_out/##AIRLOCK_ID/name          = "##AIRLOCK_NAME internal pump marker"; \
+/obj/abstract/airlock_marker/pump/internal_out/##AIRLOCK_ID/airlock_id    = "##AIRLOCK_ID"; \
+/obj/abstract/airlock_marker/pump/external_out/##AIRLOCK_ID/name          = "##AIRLOCK_NAME external pump marker"; \
+/obj/abstract/airlock_marker/pump/external_out/##AIRLOCK_ID/airlock_id    = "##AIRLOCK_ID"; \
+/obj/abstract/airlock_marker/button/interior/##AIRLOCK_ID/name            = "##AIRLOCK_NAME interior button marker"; \
+/obj/abstract/airlock_marker/button/interior/##AIRLOCK_ID/airlock_id      = "##AIRLOCK_ID"; \
+/obj/abstract/airlock_marker/button/exterior/##AIRLOCK_ID/name            = "##AIRLOCK_NAME exterior button marker"; \
+/obj/abstract/airlock_marker/button/exterior/##AIRLOCK_ID/airlock_id      = "##AIRLOCK_ID"; \
+/obj/abstract/airlock_marker/door/interior/##AIRLOCK_ID/name              = "##AIRLOCK_NAME interior door marker"; \
+/obj/abstract/airlock_marker/door/interior/##AIRLOCK_ID/airlock_id        = "##AIRLOCK_ID"; \
+/obj/abstract/airlock_marker/door/exterior/##AIRLOCK_ID/name              = "##AIRLOCK_NAME exterior door marker"; \
+/obj/abstract/airlock_marker/door/exterior/##AIRLOCK_ID/airlock_id        = "##AIRLOCK_ID";
```

Comment:
```suggestion
/obj/abstract/airlock_marker/airlock_controller/##AIRLOCK_ID/airlock_id   = #AIRLOCK_ID; \
/obj/abstract/airlock_marker/sensor/chamber/##AIRLOCK_ID/name             = #AIRLOCK_NAME + " chamber sensor marker"); \
/obj/abstract/airlock_marker/sensor/chamber/##AIRLOCK_ID/airlock_id       = #AIRLOCK_ID; \
/obj/abstract/airlock_marker/sensor/internal/##AIRLOCK_ID/name            = #AIRLOCK_NAME + " internal sensor marker"; \
/obj/abstract/airlock_marker/sensor/internal/##AIRLOCK_ID/airlock_id      = #AIRLOCK_ID; \
/obj/abstract/airlock_marker/sensor/external/##AIRLOCK_ID/name            = #AIRLOCK_NAME + " external sensor marker"; \
/obj/abstract/airlock_marker/sensor/external/##AIRLOCK_ID/airlock_id      = #AIRLOCK_ID; \
/obj/abstract/airlock_marker/pump/chamber/##AIRLOCK_ID/name               = #AIRLOCK_NAME + " chamber pump marker"; \
/obj/abstract/airlock_marker/pump/chamber/##AIRLOCK_ID/airlock_id         = #AIRLOCK_ID; \
/obj/abstract/airlock_marker/pump/internal_out/##AIRLOCK_ID/name          = #AIRLOCK_NAME + " internal pump marker"; \
/obj/abstract/airlock_marker/pump/internal_out/##AIRLOCK_ID/airlock_id    = #AIRLOCK_ID; \
/obj/abstract/airlock_marker/pump/external_out/##AIRLOCK_ID/name          = #AIRLOCK_NAME + " external pump marker"; \
/obj/abstract/airlock_marker/pump/external_out/##AIRLOCK_ID/airlock_id    = #AIRLOCK_ID; \
/obj/abstract/airlock_marker/button/interior/##AIRLOCK_ID/name            = #AIRLOCK_NAME + " interior button marker"; \
/obj/abstract/airlock_marker/button/interior/##AIRLOCK_ID/airlock_id      = #AIRLOCK_ID; \
/obj/abstract/airlock_marker/button/exterior/##AIRLOCK_ID/name            = #AIRLOCK_NAME + " exterior button marker"; \
/obj/abstract/airlock_marker/button/exterior/##AIRLOCK_ID/airlock_id      = #AIRLOCK_ID; \
/obj/abstract/airlock_marker/door/interior/##AIRLOCK_ID/name              = #AIRLOCK_NAME + " interior door marker"; \
/obj/abstract/airlock_marker/door/interior/##AIRLOCK_ID/airlock_id        = #AIRLOCK_ID; \
/obj/abstract/airlock_marker/door/exterior/##AIRLOCK_ID/name              = #AIRLOCK_NAME + " exterior door marker"; \
/obj/abstract/airlock_marker/door/exterior/##AIRLOCK_ID/airlock_id        = #AIRLOCK_ID;
```

Reply:
You have no other option. Using `##` doesn't work inside strings, and an initialize override would render it useless for mapping.
Also, that doesn't lead to string tree corruption, at least not for several years now. Every span macro on several large servers uses concatenation instead of interpolation so that it works at compile-time.